src/foolscap/banana.py

```diff
@@ -54,7 +58,7 @@ def long_to_bytes(n, blocksize=0):
     n = long(n)
     pack = struct.pack
     while n > 0:
-        s = pack('>I', n & 0xffffffffL) + s
+        s = pack('>I', n & long(0xffffffff)) + s
         n = n >> 32
```

Comment:
Do these even need to be "long"..? (On python2, I mean)

Reply:
`long_to_bytes()` was inherited from ancient PyCrypto (2006 or even earlier), which I'm sure was written before the long/int merge was even contemplated. Back then, it might have been called with either an int or a long. Maybe the bitwise-or operator didn't like being given a combination of an int and a long? Or maybe the right-shift did something weird when given a (32-bit) int instead of a long?

I suspect it was for a python bug/misbehavior that doesn't exist anymore, and certainly won't with python3. I'd agree with @meejah, just get rid of the `long()` coercions entirely.

Reply:
I usually went with a very rigorous "translate(port) your logic without modifying it" approach so kept the longs - to avoid any subtle bugs from getting in through the port. Now that this is clear, I will undo the longs. Thanks.

Reply:
Removed the long literal but kept the long(..) for time being as otherwise a test checking the type of a return value vs passed value fails in Py2. Will be doing a better long->int convergence in a later PR when we inch closer to all tests passing in Py3.